src/tildePrefix.ml

```diff
@@ -37,46 +37,123 @@
 *)
 open CST
 
-(** Remove the ['~'] character at the beginning of the given string or fail with
-    [Invalid_arg] if the string is empty or does not start with a tilde. *)
-let strip_tilde s =
-  if s = "" || s.[0] != '~' then
+(** Tests whether the given string starts with a tilde character. *)
+let starts_with_tilde string =
+  string != "" && string.[0] = '~'
+
+(** Removes the ['~'] character at the beginning of the given string or fail
+    with [Invalid_arg] if the string does not start with a tilde. *)
+let strip_tilde string =
+  if not (starts_with_tilde string) then
     invalid_arg "strip_tilde";
-  String.(sub s 1 (length s - 1))
+  String.(sub string 1 (length string - 1))
 
-(** Takes a string assumed to come from a [WordLiteral] and to start with ['~']
-    and splits it between a [WordTildePrefix] containing everything between
-    ['~'] and the first ['/'] and a [WordLiteral] containing everything else. *)
-let find_login s =
-  match String.split_on_char '/' s with
-  | login :: rem ->
-    [
-      WordTildePrefix (strip_tilde login);
-      WordLiteral (String.concat "/" rem)
-    ]
-  | _ ->
+(** Extracts the tilde-prefix at the beginning of the given literal string or
```

Comment:
NIT: Splitting on '/' to remove the first word before the first '/' seems a bit overkill. If we had a `String.split_on_char_as_seq`, this function could have been made more efficient.